HACKING.md

```diff
@@ -224,10 +224,51 @@ OpenShift is checked into this repository.  To install `godep` locally run:
 
 If you are not updating packages you should not need godep installed.
 
-## Cherry-picking an upstream commit into Origin
+## Cherry-picking an upstream commit into Origin: Why, how, and when.
+
+Upstream origin carries patches inside of vendor/ on top of each rebase.  Thus, Origin consumes kubernetes in two ways.
+
+1. Periodic rebases against a kubernetes commit.  Eventually, any patch you have in upstream kubernetes will land in openshift via this mechanism.
+
+2. Cherry picked patches for important features.  This is done when an important feature/fix is in a version of kubernetes that we have not yet rebased against.
```

Comment:
"Cherry-picked patches for important bug fixes". We're trying to limit feature backporting to 0 if possible.